content/academy/web_scraping_for_beginners/challenge/scraping_amazon.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Scraping Amazon
+description: Before you build your first web scraper with Crawlee, it is important to understand the concept of modularity in programming.
+menuWeight: 4
+paths:
+- web-scraping-for-beginners/challenge/scraping-amazon
+---
+
+# Scraping Amazon
+
+In our quick chat about modularity, we finished the code for the results page, and added a request for each product to the crawler's **RequestQueue**. Here, we just need to scrape the description, so it shouldn't be too hard:
+
+```JavaScript
+// routes.js
+
+// ...
+
+router.addHandler(labels.PRODUCT, async ({ $, crawler, request }) => {
+    const { data } = request.userData;
+
+    const element = $('div#productDescription');
+
+    const description = element.text().trim();
+
+    console.log(description); // works!
+});
+```
+
+> If you are sometimes getting an error along the lines of **RequestError: Proxy responded with 407**, don't worry, this is totally normal. The request will retry and succeed.
+
+Great! But wait, where do we go from here? We need to go to the offers page next and scrape each offer, but how can we do that? Let's take a small break from writing the scraper and open up [Proxyman]({{@link tools/proxyman.md}}) to analyze requests which we can't see inside the network tab, then we'll click the button on the product page that loads up all of the product offers:
+
+![View offers button]({{@asset web_scraping_for_beginners/challenge/images/view-offers-button.webp}})
+
+After clicking this button and checking back in Proxyman, we discovered this link:
+
+```text
+https://www.amazon.com/gp/aod/ajax/ref=auto_load_aod?asin=B07ZPKBL9V&pc=dp
+```
+
+The `asin` [query parameter](https://branch.io/glossary/query-parameters/) matches up with our product's ASIN, which means we can use this for any product of which we have the ASIN.
+
+Here's what this page looks like:
+
+![View offers page]({{@asset web_scraping_for_beginners/challenge/images/offers-page.webp}})
+
+Wow, that's ugly. But for our scenario, this is really great. When we click the **View offers** button, we usually have to wait for the offers to load and render, which would mean we could have to switch our entire crawler to a **PuppeteerCrawler** or **PlaywrightCrawler**. The data on this page we've just found appears to be loaded statically, which means we can still use CheerioCrawler and keep the scraper as efficient as possible 😎
+
+First, we'll create a function which can generate an offers URL for us in **constants.js**:
+
+```JavaScript
+// constants.js
+
+// ...
+
+export const OFFERS_URL = (asin) => `${BASE_URL}/gp/aod/ajax/ref=auto_load_aod?asin=${asin}&pc=dp`;
```

Comment:
a constant, UPPERCASE function is a really terrible thing, please get rid of it 😄